src/primitives/box.jl

```diff
@@ -33,6 +33,23 @@ measure(b::Box) = prod(b.max - b.min)
 diagonal(b::Box) = norm(b.max - b.min)
 sides(b::Box) = b.max - b.min
 
+"""
+Return all corners of the box.
+"""
+@generated function vertices(b::Box)
```

Comment:
That is awesome. As usual you playing with very fancy Julia constructs 🕶️ Can we also pass type parameters `b::Box{Dim,T}` instead of querying with `embeddim` and `coordtype` inside the function?

Reply:
Sure.